### ISSUE TITLE
Omit version number not working with spring boot

### ISSUE THREAD

**Issue by mjsaber** [spring-gradle-plugins/dependency-management-plugin#220]
Hi,

I have a multimodule project and I'm using spring boot, but looks like applying dependency without version number is not working for me.

In the root build.gradle:
```
plugins {
    id "org.springframework.boot" version "2.0.3.RELEASE" apply(false)
}

subprojects {

    apply plugin: "java"
    apply plugin: "io.spring.dependency-management"

    sourceCompatibility = JavaVersion.VERSION_1_8
    targetCompatibility = JavaVersion.VERSION_1_8


    dependencies {
        implementation "org.springframework.boot:spring-boot-starter"
    }

    repositories {
        jcenter()
    }
}
```

In one of the module the build.gradle:
```
dependencies {
}
```

And if I do `gradle demo:dependencyInsight --dependency spring-boot-starter`, I got 
```
org.springframework.boot:spring-boot-starter: FAILED
   Failures:
      - Could not find org.springframework.boot:spring-boot-starter:.

org.springframework.boot:spring-boot-starter FAILED
\--- compileClasspath
```

But if I add 
```
    dependencyManagement {
        imports {
            mavenBom 'io.spring.platform:platform-bom:Cairo-RELEASE'
        }
    }
```
I can resolve spring-boot-starter version 2.0.1.RELEASE.
Is this looking right to you? I'd expect to be able to resolve version 2.0.3.RELEASE (same as sping boot plugin version) even without specify the maven bom file.

**Comment by wilkinsona** [comment]
You haven't applied the Spring Boot plugin so it hasn't imported its bom.

**Comment by mjsaber** [comment]
@wilkinsona thanks for your reply, but apply plugin: "io.spring.dependency-management" itself is not good enough? Some of my submodules are just common util class which I don't need to apply the entire spring boot plugin.

**Comment by wilkinsona** [comment]
It's not good enough if you don't also give it a bom to use. It sounds like you probably want to do [this](https://docs.spring.io/spring-boot/docs/current/gradle-plugin/reference/html/#managing-dependencies-using-in-isolation).

**Comment by mjsaber** [comment]
@wilkinsona got it, thanks! Another related question is that does apply spring boot plugin implicitly applied dependency-management?
I have a `common module` and a `api module`.
I don't apply dependency-management plugin in both modules, and in the `common module` I don't apply spring boot plugin, and specify to use `mysql-connector-java:6.0.6`, and in the api module I include the `common module`, and apply the spring boot plugin.
When I inspect common module, I can see that the `mysql-connector-java:6.0.6`, but when I inspect the api module, I can see it being corrected to `5.1.43 (selected by rule)`.
```
mysql:mysql-connector-java:5.1.43 (selected by rule)
   variant "default" [
      org.gradle.status = release (not requested)
      Requested attributes not found in the selected variant:
         org.gradle.usage  = java-api
   ]

mysql:mysql-connector-java:6.0.6 -> 5.1.43
\--- project :messaging-platform-common
     \--- project :messaging-platform-gateway
          \--- compileClasspath
```

| module                 | spring boot plugin | dependency management plugin |
| common module  | false                       | false                                                |
| api module           | true                         | false                                                |

**Comment by wilkinsona** [comment]
The dependency management plugin isn't applied implicitly. If you apply both the dependency management plugin and Spring Boot's plugin, then Spring Boot's plugin will import its bom. There is no way to disable that behaviour short of only applying either the dependency management plugin or Spring Boot's plugin.
